### DISCUSSION TITLE
Public League Historical Data

### DISCUSSION THREAD

**Discussion by djt89** [cwendt94/espn-api#233] [ESPN API]
Hello All, 

I was wondering if there was  a way to grab all historical public league data? If not, has anyone tried to find a way to do this?

**Comment by dtcarls** [comment]
When initiating the league object use a previous year. I believe 2 or 3
years ago ESPN completely revamped their API so you might not be able to go
farther back than that.

**Comment by cwendt94** [comment]
You can go through each year your league has been active and get most of the data. Like @dtcarls said ESPN changed their API a couple years ago so some features won't work like `box_scores` however the general information is still available (Teams, Rosters, Draft, Standings).
Example
```python3
from espn_api.football import League
league_id = 1234
for i in range(2015, 2022):
   league = League(league_id, year=i)
   print(league.teams)
```

**Reply by djt89** [reply]
I am looking to get all public league data, not just for my league. Seems like this may not be possible without having to iterate through league ids until it finds ones that work.

**Reply by cwendt94** [reply]
Oh yeah that is correct. I have not found an ESPN API that allows you to get all public league's.
